modules/client/src/main/scala/lucuma/itc/client/ItcClient.scala

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.itc.client
+
+import cats.Applicative
+import cats.effect.Async
+import cats.effect.Ref
+import cats.effect.Resource
+import cats.syntax.applicativeError.*
+import cats.syntax.apply.*
+import cats.syntax.either.*
+import cats.syntax.flatMap.*
+import cats.syntax.functor.*
+import clue.TransactionalClient
+import clue.http4s.Http4sBackend
+import io.circe.syntax.*
+import lucuma.core.model.Observation
+import org.http4s.Uri
+import org.http4s.jdkhttpclient.JdkHttpClient
+import org.typelevel.log4cats.Logger
+
+/**
+ * Client for calling the ITC on the JVM.
+ */
+trait ItcClient[F[_]] {
+
+  // TODO: chart, version, etc.
+
+  def spectroscopy(
+    input:    SpectroscopyModeInput,
+    useCache: Boolean = true
+  ): F[Either[Throwable, List[SpectroscopyResult]]]
+
+}
+
+object ItcClient {
+
+  def create[F[_]: Async: Logger](
+    uri: Uri
+  ): F[ItcClient[F]] =
+    Ref
+      .of[F, Map[SpectroscopyModeInput, Either[Throwable, List[SpectroscopyResult]]]](Map.empty)
+      .map { cache =>
+        // TODO: Cache contains failed results and is not flushed until the next server restart.
+        // TOOD: Likely we don't want to cache failures and should flush at least when the version changes.
```

Comment:
We may want to turn the cache into an interface we can replace with a distributed cache at some point.